datastream/tools/split_dataframe.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+from typing import Tuple, Union, Dict, Optional
+from pathlib import Path
+import numpy as np
+import pandas as pd
+
+from datastream import json
+
+
+def split_dataframe(
+    dataframe: pd.DataFrame,
+    key_column: str,
+    proportions: Dict[str, float],
+    filepath: Optional[Path] = None,
+    stratify_column: Optional[str] = None,
+):
+    '''
+    Split and save result. Add new examples and continue from the old split.
+
+    As new examples come in it can handle:
+    - Changing test size
+    - Adapt after removing examples from dataset
+    - Adapt to new stratification
+    '''
+    if abs(sum(proportions.values()) - 1.0) >= 1e-6:
```

Comment:
how sensitive is this 1e-6?
https://stackoverflow.com/a/19141711/905596
if you close to 10 proportions and are running on 32 bit system you might stumble on problems

Reply:
I think 10 proportions is an unrealistic use-case? Then again, I suppose you could use this to set up k-folds so maybe not? Anyway, it seems unlikely to me that someone would be doing deep learning on a 32-bit system, no?

Reply:
We can lower 1e-6 to 1e-5